### PR TITLE
Added metro configuration to allow developing in the example folder

### PR DESCRIPTION
**Pull request by rewieer**
As discussed in #261 it would be great to develop in the Example folder as to enhance our maintainer experience.

After a fair bit of work, I managed to make it work. Dependencies are updated to RN 0.58 in the example folder, but the project have been tested into a RN 0.57.1 project (Android) so i'm pretty much confident this will work. Unfortunately i've been unable to make metro work with RN56 nor RN57, so I updated them to RN58. The documentation is far from being complete.

Looking for some feedback and tests on this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,13 +30,17 @@
   },
   "dependencies": {
     "prop-types": "^15.6.2",
-    "react-native-animatable": "^1.2.4"
+    "react-native-animatable": "1.3.1"
```
Comment (mmazzarolo) [inline package.json]: I would move this change to a separate PR, since it might have a bigger impact than we expect (so it needs to be tested carefully).

File: .gitignore [hunk 2]
```diff
@@ -7,4 +7,7 @@ node_modules/
 npm-debug.log
 jsconfig.json
 dump.rdb
-.vscode
\ No newline at end of file
+.vscode
+.idea
+yarn.lock
+example/yarn.lock
```
Comment (mmazzarolo) [inline .gitignore]: 👍

File: example/rn-cli.config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+const reactNativeModalRoot = path.resolve(__dirname, '..');
+
+module.exports = {
+  projectRoot: __dirname,
+  watchFolders: [reactNativeModalRoot],
+  resolver: {
+    blacklistRE: blacklist([new RegExp(`${reactNativeModalRoot}/node_modules/react-native/.*`)]),
+    providesModuleNodeModules: ['react-native-animatable']
+  },
+};
```
Comment (mmazzarolo) [inline example/rn-cli.config.js]: Neat!

File: .gitignore (hunk 2 above)
Comment (rewieer) [reply .gitignore]: Though I think we should use yarn. I did use npm but to update that package-lock.json it took me 20 seconds more

File: package.json (hunk 1 above)
Comment (rewieer) [reply package.json]: This is an error actually, we don't need to upgrade react-native-animatable for this to work so i'm definetely going to revert this 😄

Comment (rewieer) [reply package.json]: @mmazzarolo updated !

**Comment by mmazzarolo** [comment]
Available in [v10.0.0-next.0](
https://github.com/react-native-community/react-native-modal/releases/tag/v10.0.0-next.0) ✈️
